src/components/navigation-menu/navigation-menu.tsx

```diff
@@ -0,0 +1,74 @@
+import classNames from 'classnames';
+import { NavLink } from '@remix-run/react';
+import { ROUTES } from '~/router/config';
+import { CategoryLink } from '../category-link/category-link';
+import styles from './navigation-menu.module.scss';
+
+interface NavigationMenuProps {
+    className?: string;
+    vertical?: boolean;
+    onLinkClick?: () => void;
+}
+
+export const NavigationMenu = ({
+    className,
+    vertical = false,
+    onLinkClick,
+}: NavigationMenuProps) => {
+    const menuItemStyle = ({ isActive }: { isActive: boolean }) =>
+        classNames(styles.menuItem, {
+            [styles.active]: isActive,
+        });
+
+    return (
+        <nav className={className}>
+            <ul className={classNames(styles.menuList, { [styles.vertical]: vertical })}>
+                <li>
+                    <CategoryLink
+                        categorySlug="all-products"
+                        className={menuItemStyle}
+                        onClick={onLinkClick}
+                    >
+                        Shop All
+                    </CategoryLink>
+                </li>
+                <li>
+                    <CategoryLink
+                        categorySlug="kitchen-essentials"
+                        className={menuItemStyle}
+                        onClick={onLinkClick}
```

Comment:
Does remix give us a hook something like 'onNavigate'? Or maybe we can add a single click handler on the root and check if event target is a link. So if the user wants to add/remove an item they wouldn't need to wire it to onClick.